### PR TITLE
Update Databricks doc to include instructions on how to submit Spark .NET job with UDFs

### PR DESCRIPTION
**Pull request by Niharikadutta**
## Summary

Update Databricks doc to include instructions on how to submit Spark .NET job with UDFs

[Preview Link](https://review.docs.microsoft.com/en-us/dotnet/spark/tutorials/databricks-deployment?branch=pr-en-us-23764)

### CODE REVIEW

**Comment by Niharikadutta** [comment]
@luisquintanilla Can you please take a look at this PR?

File: docs/spark/tutorials/databricks-deployment.md [hunk 1]
```diff
@@ -143,6 +143,25 @@ Next, you publish the *mySparkApp* created in the [.NET for Apache Spark - Get S
...
+
+    Uncomment and modify the app dependencies section in [db-init.sh](https://github.com/dotnet/spark/blob/master/deployment/db-init.sh) to point to your app dependencies path. Then, upload the updated *db-init.sh* to your cluster:
+
+    ```console
+    cd <path-to-db-init-and-install-worker>
+    databricks fs cp db-init.sh dbfs:/spark-dotnet/db-init.sh
+    ```
+
+> [!Note]
+> You can find more information on how to submit a .NET for Apache Spark job to Databricks in [this article](https://docs.microsoft.com/dotnet/spark/how-to-guides/databricks-deploy-methods).
```
Comment (luisquintanilla) [inline docs/spark/tutorials/databricks-deployment.md:163]:
```suggestion
> For more information, see the [Submit a .NET for Apache Spark job to Databricks guide](https://docs.microsoft.com/dotnet/spark/how-to-guides/databricks-deploy-methods).
```

File: docs/spark/tutorials/databricks-deployment.md [hunk 2]
```diff
@@ -143,6 +143,25 @@ Next, you publish the *mySparkApp* created in the [.NET for Apache Spark - Get S
    zip -r publish.zip .
    ```
 
+3. If you have any user-defined functions in your app, the app assemblies, such as DLLs that contain user-defined functions along with their dependencies, need to be placed in the working directory of each *Microsoft.Spark.Worker*.
+
+    Upload your application assemblies to your Databricks cluster:
+
+    ```console
+    cd <path-to-your-app-publish-directory>
+    databricks fs cp <assembly>.dll dbfs:/apps/dependencies
```
Comment (luisquintanilla) [inline docs/spark/tutorials/databricks-deployment.md:152]: Is the `apps/dependencies` directory a default in Databricks or is it something that users have to create? If the latter, it'd be good to mention that.

**Review by luisquintanilla — COMMENTED** [review]
@Niharikadutta looks great. Added a few comments.

**Review by luisquintanilla — APPROVED** [review]

**Pull request merged**